cmd/quadlet/main.go

```diff
@@ -210,6 +210,37 @@ func enableServiceFile(outputPath string, service *parser.UnitFile) {
 	}
 }
 
+// warns if input is a "short name".  A "short name" refers to a
+// container image without a fully-qualified reference, and is hence
+// missing a registry (or domain).  Names including a digest are not
+// short names.
+//
+// Examples:
+//   - short names: "image:tag", "library/fedora"
+//   - not short names: "quay.io/image", "localhost/image:tag",
+//     "server.org:5000/lib/image", "image@sha256:...
+//
+// We implement a simple version of this from scratch here to avoid
+// a huge dependency in the generator just for a warning.
+func warnIfShortName(container *parser.UnitFile) {
+	imageName, ok := container.Lookup(quadlet.ContainerGroup, quadlet.KeyImage)
+	if !ok {
+		return
+	}
+	if strings.Contains(imageName, "@") {
+		return // digest, not short
+	}
+	firstSlash := strings.Index(imageName, "/")
+	if firstSlash >= 0 {
+		registry := imageName[:firstSlash]
+		if strings.ContainsAny(registry, ".:") || registry == "localhost" {
+			return // just path, not registry
+		}
+	}
+
+	Logf("Warning: %s specifies the image \"%s\" which is a short name, ideally use a fully qualified image name", container.Filename, imageName)
```

Comment:
Just to be sure: using a short name may even error out if there's no matching alias and no user-interactive session such as in a systemd unit.

Reply:
Well, if you specify a non-existing fq name it may error out too if that doesn't exist. But yes, clearly this warning could be better.

Reply:
> Well, if you specify a non-existing fq name it may error out too if that doesn't exist

That's a different issue.  Podman may error out _while_ resolving the short name. See "What happens if I run Podman in a script and cannot be prompted?" in https://www.redhat.com/sysadmin/container-image-short-names.